### DISCUSSION TITLE
where is the code for... connection with lichess

### DISCUSSION THREAD

**Discussion by 9acca9** [EdNekebno/DGTCentaurMods#146] [General]
Sorry, the dumb question, but i cant find the code related to moves, connection with lichess, etc.
Thanks.

**Comment by EdNekebno** [comment]
This depends on what you're doing. The game state is handled by gamemanager ( https://github.com/EdNekebno/DGTCentaurMods/blob/master/DGTCentaurMods/opt/DGTCentaurMods/game/gamemanager.py ) . A script using it can pass in some functions to get keypresses, events (such as which turn it is, a new game, game over, etc), and moves. Example code for this callback system is here: https://github.com/EdNekebno/DGTCentaurMods/blob/master/DGTCentaurMods/opt/DGTCentaurMods/examples/gamemanagerex.py . Then, for example, in lichess.py ( https://github.com/EdNekebno/DGTCentaurMods/blob/master/DGTCentaurMods/opt/DGTCentaurMods/game/lichess.py ) there is a movecallback function on line 144 which calls the berserk library to make a move on line 159.

**Comment by 9acca9** [comment]
Thanks for the answer!!!
